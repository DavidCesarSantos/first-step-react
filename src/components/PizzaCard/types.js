import { exact, string, number } from "prop-types";

export const pizza = exact({
  id: string,
  title: string,
  description: string,
  price: number,
  currency: string,
  image: exact({
    url: string
  })
});
