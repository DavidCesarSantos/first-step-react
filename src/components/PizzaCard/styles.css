.pizza-card-media {
  height: 200px;
}
