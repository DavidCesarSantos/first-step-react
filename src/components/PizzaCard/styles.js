import glamorous from "glamorous";
import { CardMedia } from "material-ui";

export const PizzaCardMedia = glamorous(CardMedia)({
  height: "200px"
});
