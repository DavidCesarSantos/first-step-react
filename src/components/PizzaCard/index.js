import React, { Component } from "react";
import Button from "material-ui/Button";
import Card, { CardActions, CardContent, CardMedia } from "material-ui/Card";
import Typography from "material-ui/Typography";
import { func } from "prop-types";

import Price from "../Price";
import { pizza } from "./types";

import "./styles.css";

export default class Product extends Component {
  static propTypes = {
    details: pizza,
    addOrder: func
  };

  constructor(props) {
    super(props);

    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.props.addOrder(this.props.details.id);
  }

  render() {
    const { title, description, price, currency, image } = this.props.details;
    return (
      <Card>
        <CardMedia
          image={image.url}
          title={title}
          className="pizza-card-media"
        />
        <CardContent>
          <Typography variant="headline" component="h4">
            {title}
          </Typography>
          <Typography component="p">{description}</Typography>
          <Price price={price} currency={currency} />
        </CardContent>
        <CardActions>
          <Button size="small" color="primary" onClick={this.onClick}>
            Ajouter
          </Button>
        </CardActions>
      </Card>
    );
  }
}
