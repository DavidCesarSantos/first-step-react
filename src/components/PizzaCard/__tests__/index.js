import React, { Component } from "react";
import renderer from "react-test-renderer";

import PizzaCard from "../";

const details = {
  id: "2",
  title: "FETA & MIEL - LARDONS",
  description: "Base crème fraîche, mozzarella, lardons fumés, feta, miel doux",
  price: 1199,
  currency: "€",
  image: {
    url:
      "http://thumb.nespi.1mag.kz/A6SzcsAbTHfti_2Jc1p1f1gWO78=/250x0/smart/http://nespi.kz/media/places/photos/c6/7f/a6/12/c67fa61255d756a710cd3a7218529708.jpg"
  }
};
const mutations = [
  {
    name: "renders with addOrder",
    component: PizzaCard,
    props: {
      details: details
    }
  }
];

mutations.forEach(mutation => {
  it(`renders ${mutation.name}`, () => {
    const myComponent = renderer.create(
      <mutation.component {...mutation.props} />
    );
    expect(myComponent.toJSON()).toMatchSnapshot();
  });
});

// it("renders without crashing", () => {
//   const details = {
//     id: "2",
//     title: "FETA & MIEL - LARDONS",
//     description:
//       "Base crème fraîche, mozzarella, lardons fumés, feta, miel doux",
//     price: 1199,
//     currency: "€",
//     image: {
//       url:
//         "http://thumb.nespi.1mag.kz/A6SzcsAbTHfti_2Jc1p1f1gWO78=/250x0/smart/http://nespi.kz/media/places/photos/c6/7f/a6/12/c67fa61255d756a710cd3a7218529708.jpg"
//     }
//   };
//   const myComponent = renderer.create(<PizzaCard details={details} />);
//   expect(myComponent.toJSON()).toMatchSnapshot();
// });

// it("renders with addOrder", () => {
//   const myComponent = <PizzaCard addOrder={() => {}} />;
//   expect(myComponent).toMatchSnapshot();
// });

// it("renders with details", () => {
//   const details = {
//     id: "2",
//     title: "FETA & MIEL - LARDONS",
//     description:
//       "Base crème fraîche, mozzarella, lardons fumés, feta, miel doux",
//     price: 1199,
//     currency: "€",
//     image: {
//       url:
//         "http://thumb.nespi.1mag.kz/A6SzcsAbTHfti_2Jc1p1f1gWO78=/250x0/smart/http://nespi.kz/media/places/photos/c6/7f/a6/12/c67fa61255d756a710cd3a7218529708.jpg"
//     }
//   };
//   const myComponent = <PizzaCard details={details} />;
//   expect(myComponent).toMatchSnapshot();
// });
