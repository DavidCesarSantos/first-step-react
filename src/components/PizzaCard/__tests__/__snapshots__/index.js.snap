// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders with addOrder 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="MuiCardMedia-root-28 pizza-card-media"
    src={undefined}
    style={
      Object {
        "backgroundImage": "url(http://thumb.nespi.1mag.kz/A6SzcsAbTHfti_2Jc1p1f1gWO78=/250x0/smart/http://nespi.kz/media/places/photos/c6/7f/a6/12/c67fa61255d756a710cd3a7218529708.jpg)",
      }
    }
    title="FETA & MIEL - LARDONS"
  />
  <div
    className="MuiCardContent-root-30"
  >
    <h4
      className="MuiTypography-root-31 MuiTypography-headline-36"
    >
      FETA & MIEL - LARDONS
    </h4>
    <p
      className="MuiTypography-root-31 MuiTypography-body1-40"
    >
      Base crème fraîche, mozzarella, lardons fumés, feta, miel doux
    </p>
    <p
      className="MuiTypography-root-31 MuiTypography-body1-40"
    >
      11.99
      €
    </p>
  </div>
  <div
    className="MuiCardActions-root-55"
  >
    <button
      className="MuiButtonBase-root-72 MuiButton-root-57 MuiButton-flatPrimary-59 MuiButton-sizeSmall-69 MuiCardActions-action-56"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-58"
      >
        Ajouter
      </span>
      <span
        className="MuiTouchRipple-root-75"
      />
    </button>
  </div>
</div>
`;
